README.md

```diff
@@ -1,42 +1,143 @@
+* [Introduction](#Introduction)
+* [1. Images](#1-images)
+  * [1.1 How circleci uses our images in testing](#11-How-circleci-uses-our-images-in-testing)
+* [2. How to update images](#2-How-to-update-images)
+  * [2.1 How to update a postgres minor version in images](#2.1-How-to-update-postgres-minor-version-in-images)
+  * [2.2 How to add a major postgres major](#2.2-How-to-add-a-major-postgres-major)
+* [3. How to publish a change](#3-How-to-publish-a-change)
+
+## Introduction
+
+This repository contains the source code for docker images that are used in [citus testing](https://github.com/citusdata/citus/blob/master/.circleci/config.yml). The images are pushed to [docker hub](https://hub.docker.com/u/citus). There is no hooking logic between this repository and the docker hub account. It is used purely for the storage of our images source code.
+
+## 1. Images
 
 The [extbuilder](https://github.com/citusdata/the-process/tree/master/circleci/images/extbuilder) image is the core image that other jobs depend on in our tests. The [extbuilder](https://github.com/citusdata/the-process/tree/master/circleci/images/extbuilder):
 
-- Installs postgres 10 and 11 `pg_config`. We only need `pg_config` to generate citus artifacts.
-- Installs and tars the checked out citus version for the postgres versions. This is done so that the other jobs can install citus easily by untarring without the need to do `make install`.
-- Creates `build-{pg_version}` folders with citus configured, so these folders have the necessary `Makefile.global` to run the tests. These folders are generated for each postgres version. For example, at the time of writing there are 2 folders with `build-10` and `build-11`.
+While building the image:
+
+* Installs postgres majors' `pg_config`. We only need `pg_config` to generate citus artifacts.
+
+While running the container:
+
+* Installs and tars the checked out citus version for the postgres versions. This is done so that the other jobs can install citus easily by untarring without the need to do `make install`. We need to do this step in the running container, not while building the image, so that the built citus version is the checked out citus code.
+* Creates `build-{pg_version}` folders with citus configured, so these folders have the necessary `Makefile.global` to run the tests. These folders are generated for each postgres version. For example, at the time of writing there are 3 folders with `build-10`, `build-11` and `build-12`.
 
 The general process for other images are similar, which is:
 
-- Untar the citus tar to install citus for the postgres version that the image will use. For example, if postgres 11 is used, the tar is named `install-11.tar`.
-- The script (`install-and-test-ext`) takes the target name as an argument, and the following command is used to run the test:
+While building the image:
+
+* Install postgres major which is taken as an argument `PG_MAJOR`.
```

Comment:
most images would and should be taken from the original postgres releases, I think this line does not reflect that.

Reply:
Well if we install pg12 from the postgres base image, when we try to install `libcurl`, it installs `libcurl4` and citus does not want `libcurl4` but `libcurl3`. We can add `libcurl3` package source to our `apt/source.list` but in the future another package can cause a problem too. And in the `extbuilder` image we also dont install from pg base image as we have multiple pg majors. So installing the same way as `extbuilder` seemed okay to me, what do you think?